soroban-sdk/src/accounts.rs

```diff
@@ -111,16 +111,30 @@ impl PartialOrd for AccountId {
 
 impl Ord for AccountId {
     fn cmp(&self, other: &Self) -> core::cmp::Ordering {
-        self.0.cmp(&other.0)
+        self.0.env.check_same_env(&other.0.env);
+        let v = self
+            .0
+            .env
+            .obj_cmp(self.0.obj.to_raw(), other.0.obj.to_raw());
+        if v == 0 {
+            Ordering::Equal
+        } else if v < 0 {
+            Ordering::Less
+        } else {
+            Ordering::Greater
+        }
```

Comment:
I believe this is the same thing:
```suggestion
        v.cmp(&0)
```